ConwaysGameOfLife.Tests/GameControllerShould.cs

```diff
@@ -59,72 +59,87 @@ public void ReturnUniverseString_GivenInvalidInputsFollowedByValidInput(IEnumera
         {
             _input.GetReadLine(invalidInputs);
             _input.GetReadLine(validInput);
-            var expectedOutputLine1 = "Invalid input. Please try again.";
-            var expectedOutputLine2 = "Please enter the width & length for the Universe as a number followed by a comma then a number (e.g. '0,0').";
+            var invalidOutputMessageLine1 = "Invalid input. Please try again.";
+            var invalidOutputMessageLine2 = "Please enter the width & length for the Universe as a number followed by a comma then a number (e.g. '0,0').";
+            var count = invalidInputs.Count();
 
             var result = _controller.CreateValidUniverseString();
-            var actualOutputLine1 = _output.GetWriteLine(0);
-            var actualOutputLine2 = _output.GetWriteLine(1);
+
+            for (var i = 0; i <= count; i += 2)

```

Comment:
Think this will only loop through half of the invalid inputs
```suggestion
            for (var i = 0; i < count * 2; i += 2)
```

Reply:
ahh that's my bad, good pick up @Kedoodle ! I think at the moment this test is focusing too much on the implementation details and it would be enough to give it all the invalid inputs followed by the valid input, then check that the output is equal to the valid input. This way, it would inherently test that the method will stop expecting input only until it receives a valid output.